lib/scripts.js

```diff
@@ -305,93 +311,51 @@ var scripts = {
    *
    * @param {Queue} queue The queue for the job
    * @param {Job} job The job
-   * @param {Boolean=false} renew Whether to renew to lock, meaning it will assume the job 
+   * @param {Boolean=false} renew Whether to renew to lock, meaning it will assume the job
    *    is already locked and just reset the lock expiration.
    * @param {Boolean=false} ensureActive Ensures that the job is in the 'active' state.
    */
-  takeLock: function(queue, job, token, renew, ensureActive){
-    // Ensures that the lock doesn't exist, or if it does, that we own it.
-    var ensureOwnershipCall = [
-      'local prevLock = redis.call("GET", KEYS[1])',
-      'if (prevLock and prevLock ~= ARGV[1]) then',
-      '  return 0',
-      'end'
-    ].join('\n');
-
-    // Ensures that the lock in the 'active' state.
-    var ensureActiveCall = [
-      // Note that while this is inefficient to run a O(n) traversal of the 'active' queue,
-      // it's highly likely that the job is within the first few elements of the active
-      // list at the time this call is used.
-      'local activeJobs = redis.call("LRANGE", KEYS[3], 0, -1)',
-      'local found = false',
-      'for _, job in ipairs(activeJobs) do',
-      '  if(job == ARGV[3]) then',
-      '     found = true',
-      '     break',
-      '  end',
-      'end',
-      'if (found == false) then',
-      '  return 0',
-      'end'
-    ].join('\n');
+  takeLock: function(queue, job, renew, ensureActive){
+    var lock = job.lock;
+    if (renew && !lock) {
+      throw new Error('Unable to renew nonexisting lock');
+    } else if (renew) {
+      return lock.extend(queue.LOCK_RENEW_TIME);
```

Comment:
The [redlock extend method uses PEXPIRE](https://github.com/nextorigin/node-redlock/blob/master/redlock.js#L16), and should be functionally equivalent to the SET ... PX method bull uses now.  It only sets the expire from the current time, it does not add to the existing TTL, so it's typical redis expire behavior.